pkg/liqo-controller-manager/authentication/identitycreator-controller/identitycreator_controller.go

```diff
@@ -0,0 +1,136 @@
+	client.Client
+	*runtime.Scheme
+
+	eventRecorder record.EventRecorder
+
+	liqoNamespace        string
+	localClusterIdentity *discoveryv1alpha1.ClusterIdentity
+}
+
+// cluster-role
+// +kubebuilder:rbac:groups=authentication.liqo.io,resources=resourceslices;resourceslices/status,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=authentication.liqo.io,resources=identities,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile local ResourceSlice resources.
+func (r *IdentityCreatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var resourceSlice authv1alpha1.ResourceSlice
+	if err := r.Get(ctx, req.NamespacedName, &resourceSlice); err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(4).Infof("resourceSlice %q not found", req.NamespacedName)
+			return ctrl.Result{}, nil
+		}
+		klog.Errorf("unable to get ResourceSlice %q: %v", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	}
+
+	if !r.resourceSliceAccepted(&resourceSlice) {
+		klog.V(4).Infof("Skip creation of identity as resourceslice %q is not valid or denied", req.NamespacedName)
+		r.eventRecorder.Event(&resourceSlice, corev1.EventTypeWarning, "IdentitySkipped", "Identity skipped as ResourceSlices not accepted")
+		return ctrl.Result{}, nil
+	}
+
+	// TODO: should we cancel the Identity if status condition become Denied?
+
+	// Create or update the Identity resource.
+	identity := forge.Identity(forge.ResourceSliceIdentityName(&resourceSlice))
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, identity, func() error {
+		forge.MutateIdentity(identity, *resourceSlice.Spec.ProviderClusterIdentity, authv1alpha1.VirtualNodeIdentityType,
+			resourceSlice.Status.AuthParams, nil)
+		return controllerutil.SetOwnerReference(&resourceSlice, identity, r.Scheme)
+	}); err != nil {
+		klog.Errorf("unable to create or update Identity %q: %v", identity.Name, err)
+		return ctrl.Result{}, err
+	}
+
+	klog.Infof("Ensured Identity associated to ResourceSlice %q", req.NamespacedName)
+	r.eventRecorder.Event(&resourceSlice, corev1.EventTypeNormal, "IdentityEnsured", "Identity ensured")
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *IdentityCreatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// generate the predicate to filter just the ResourceSlices created by the local cluster checking crdReplicator labels
+	localResSliceFilter, err := predicate.LabelSelectorPredicate(reflection.LocalResourcesLabelSelector())
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&authv1alpha1.ResourceSlice{}, builder.WithPredicates(localResSliceFilter), r.withResourceSliceAccepted()).
+		Complete(r)
+}
+
+func (r *IdentityCreatorReconciler) withResourceSliceAccepted() builder.ForOption {
```

Comment:
In the predicate, I would only check that the condition is set, and not to check its state letting us to react on denied resources